### PR TITLE
Use a thread priority library from realtime_tools

### PR DESCRIPTION
**Pull request by AndyZe**
Depends on this `realtime_tools` PR:  https://github.com/ros-controls/realtime_tools/pull/83

### CODE REVIEW

File: controller_manager/CMakeLists.txt [hunk 1]
```diff
@@ -21,10 +22,9 @@ foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
   find_package(${Dependency} REQUIRED)
 endforeach()
 
-add_library(${PROJECT_NAME} SHARED
-  src/controller_manager.cpp
-  src/realtime.cpp
-)
+include_directories(${realtime_tools_INCLUDE_DIRS})
+
+add_library(${PROJECT_NAME} SHARED src/controller_manager.cpp)
```
Comment (DasRoteSkelett) [inline controller_manager/CMakeLists.txt:27]: For a better diff, I would prefer
```
add_library(${PROJECT_NAME} SHARED
  src/controller_manager.cpp
)
```

**Review by DasRoteSkelett — APPROVED** [review]
lgtm :)

File: controller_manager/src/ros2_control_node.cpp [hunk 2]
```diff
@@ -19,8 +19,8 @@
 #include <thread>
 
 #include "controller_manager/controller_manager.hpp"
-#include "controller_manager/realtime.hpp"
 #include "rclcpp/rclcpp.hpp"
+#include "realtime_tools/thread_priority.hpp"
```
[the remark this answers is not in the document]
Comment (AndyZe) [reply controller_manager/src/ros2_control_node.cpp:23]: That's what I prefer, too

File: controller_manager/CMakeLists.txt [hunk 3]
```diff
@@ -33,7 +33,7 @@ target_compile_definitions(${PROJECT_NAME} PRIVATE "CONTROLLER_MANAGER_BUILDING_
 
 add_executable(ros2_control_node src/ros2_control_node.cpp)
 target_include_directories(ros2_control_node PRIVATE include)
-target_link_libraries(ros2_control_node ${PROJECT_NAME})
+target_link_libraries(ros2_control_node ${PROJECT_NAME} ${realtime_tools_LIBRARIES})
```
[the remark this answers is not in the document]
Comment (AndyZe) [reply controller_manager/CMakeLists.txt:36]: done

File: controller_manager/CMakeLists.txt (hunk 1 above)
Comment (AndyZe) [reply controller_manager/CMakeLists.txt:27]: done

**Review by rosterloh — APPROVED** [review]

File: controller_manager/CMakeLists.txt [hunk 4]
```diff
@@ -33,7 +35,10 @@ target_compile_definitions(${PROJECT_NAME} PRIVATE "CONTROLLER_MANAGER_BUILDING_
 
 add_executable(ros2_control_node src/ros2_control_node.cpp)
 target_include_directories(ros2_control_node PRIVATE include)
-target_link_libraries(ros2_control_node ${PROJECT_NAME})
+target_link_libraries(ros2_control_node
+  ${PROJECT_NAME}
+  ${realtime_tools_LIBRARIES}
```
Comment (destogl) [inline controller_manager/CMakeLists.txt:40]: This should not be necessary.

File: controller_manager/CMakeLists.txt [hunk 5]
```diff
@@ -21,9 +22,10 @@ foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
   find_package(${Dependency} REQUIRED)
 endforeach()
 
+include_directories(${realtime_tools_INCLUDE_DIRS})
+
```
Comment (destogl) [inline controller_manager/CMakeLists.txt:26]: This should not be necessary.

```suggestion
```

File: controller_manager/CMakeLists.txt (hunk 4 above)
Comment (destogl) [reply controller_manager/CMakeLists.txt:40]:
```suggestion
```

File: controller_manager/src/ros2_control_node.cpp [hunk 6]
```diff
@@ -19,8 +19,8 @@
 #include <thread>
 
 #include "controller_manager/controller_manager.hpp"
-#include "controller_manager/realtime.hpp"
-#include "rclcpp/rclcpp.hpp"
+#include <rclcpp/rclcpp.hpp>
+#include <realtime_tools/thread_priority.hpp>
```
Comment (destogl) [inline controller_manager/src/ros2_control_node.cpp:23]: We are using `""` for including cpp headers.

```suggestion
#include "rclcpp/rclcpp.hpp"
#include "realtime_tools/thread_priority.hpp"
```

**Review by destogl — CHANGES_REQUESTED** [review]

File: controller_manager/src/ros2_control_node.cpp (hunk 6 above)
Comment (AndyZe) [reply controller_manager/src/ros2_control_node.cpp:23]: If that's the convention, that's fine and I'll make the change.

FYI, `"` is usually used for includes within the package. `<` is usually used for standard, system libraries. It has to do with which directories are searched first.

https://stackoverflow.com/a/21594

**Comment by bmagyar** [comment]
gears are turning slowly

humble: https://github.com/ros/rosdistro/pull/35371 -- can backport to humble branch once merged
rolling: https://github.com/ros/rosdistro/pull/35372 -- can merge once this PR once merged

**Review by bmagyar — APPROVED** [review]

**Comment by AndyZe** [comment]
Thanks for following up. 👍

**Comment by bmagyar** [comment]
No worries! Humble and rolling main will keep failing as the sync hasn't happened on those yet but the rest should be fine

**Comment by bmagyar** [comment]
@mergifyio backport to humble

**Pull request merged**